app/utils/entity_utils.rb

```diff
@@ -410,6 +410,9 @@ def error_msg(result)
     end
 
     def with_result(specs:, data:, on_success:, on_failure:)
+      if data.is_a?(ActionController::Parameters)
```

Comment:
Ideally, it would be great to keep the EntityUtils to work with Hashes only, and keep it unaware of Rails. So that it could be one day extracted to a library. So ideally, I'd move the call to `.to_unsafe_hash` to the caller side. However, I understand that it would require a lot of work to hunt down all the places where `build` or `validate` is called with `Parameters`. So adding the call here makes sense in practice.

What if we add a deprecation warning? Something like:

```
if data.is_a?(ActionController::Parameters)
  ActiveSupport::Deprecation.warn("EntityUtils used with ActionController::Parameters. Convert Parameters to Hash using `.to_unsafe_hash` before using EntityUtils")
  data = data.to_unsafe_hash
end
```